Automattic/jetpack
# Jetpack Sync: Ensure HPOS order status is prefixed with 'wc-' before sending it to WPCOM

Jetpack Sync: Ensure HPOS order status is prefixed with `wc-` before sending it to WPCOM

This is needed so that when we insert custom order statuses in the corresponding `wc_orders` table on WPCOM, we get the proper `wc-` prefix. We are handling the issue in the Sync package end since `wc_get_order_statuses` is not available when the corresponding Sync action is processed.

## Proposed changes:

* ` Automattic\Jetpack\Sync\Modules\WooCommerce_HPOS_Orders`: Introduce a wrapper for `wc_get_order_statuses` in case it's not present, but also to handle a bug in the WooCommerce codebase, since `wc-checkout-draft` is [missing](https://github.com/Automattic/woocommerce/blob/7cd90c744da9fd57b5bfb6dc489edc392e287400/plugins/woocommerce/includes/wc-order-functions.php#L96)
* ` Automattic\Jetpack\Sync\Modules\WooCommerce_HPOS_Orders`: Update `filter_order_data` to prefix the order status with `wc-` when applicable before sending the order to WPCOM

### Other information:


## Jetpack product discussion


p9dueE-8ec-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Since `get_objects_by_id` is also affected it would be nice to repeat the test instructions from https://github.com/Automattic/jetpack/pull/38251
* Add a [custom order status](https://www.cloudways.com/blog/create-woocommerce-custom-order-status/) with `exclude_from_search` set to `true` and ensure the corresponding order will be synced with the status prefixed with `wc-`.
* Ensure we are NOT adding the `wc-` prefix for  `auto-draft` and `trash` orders
